test/backdoor/README.md

```diff
@@ -0,0 +1,899 @@
+    vm.assume(to == address(0xaaaa0007));
+    vm.assume(bytes4(data) == bytes4(keccak256("handle_delegatecall()")));
+    ...
+```
+### Regular symbolic memory offset
+There is also the usual `call` by symbolic calldata:
+```solidity
+function execute(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 txGas
+    ) internal returns (bool success) {
+...
+   else {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            success := call(txGas, to, value, add(data, 0x20), mload(data), 0, 0)
+        }
+    }
+}
+```
+We deal with it as usual:
+```solidity
+...
+} else {
+   address target = svm.createAddress("execute_target");
+    bytes memory mydata;
+    //Get some concrete target-name pair
+    (target, mydata) = glob.get_concrete_from_symbolic_optimized(target);
+    target.call(mydata);
+    
+    /*assembly {
+        success := call(txGas, to, value, add(data, 0x20), mload(data), 0, 0)
+    }*/
+}
+...
+```
+Another place with symbolic offset is signature checks in `Safe::execTransaction`:
+```solidity
+function execTransaction(
+...
+bytes calldata data,
+...
+bytes memory signatures
+) public payable virtual returns (bool success) {
+...
+ bytes32 txHash;
+// Use scope here to limit variable lifetime and prevent `stack too deep` errors
+{
+    bytes memory txHashData = encodeTransactionData(
+        // Transaction info
+        to,
+        value,
+        data,
+        operation,
+        safeTxGas,
+        // Payment info
+        baseGas,
+        gasPrice,
+        gasToken,
+        refundReceiver,
+        // Signature info
+        nonce
+    );
+    // Increase nonce and execute transaction.
+    nonce++;
+    txHash = keccak256(txHashData);
+    checkSignatures(txHash, txHashData, signatures);
+...
+}
+```
+Let's handle this as we do with other cryptographic checks: we simply remove this logic, assuming that the data was entered correctly.
+### OwnerIsNotABeneficiary issue
+Before considering this problem, it is worth clarifying that it is relevant for relatively weak PCs. As practice shows, the same test on a powerful PC with a better CPU does not show this problem. Looking ahead, the problem is related to the complexity of solving computational problems and, as a result, the unstable behavior of Halmos in such cases.
+
+Let's try running the test now:
+```javascript
+$ halmos --solver-timeout-assertion 0 --function check_backdoor --loop 100
+...
+CALL SafeProxyFactory::createProxyWithCallback(...)
+...
+    CALL SafeProxy::symbolic_fallback(...)
+    ...
+        DELEGATECALL SafeProxy::setup(...)
+    ...
+    CALL  0xaaaa0006::proxyCreated(...)
+    ...
+        REVERT OwnerIsNotABeneficiary ((error: Revert())
+    ...
+```
+This revert is worth our attention, because at the moment all the code of this function that follows this revert is blocked. This revert happens 100% of the time if it reaches this point.
```

Comment:
catching up with this, are you certain this happens 100% of the time? I could see why _some_ paths would revert with this (if the `--solver-timeout-branching` is too low, then that path is not automatically eliminated so we need to consider it). But I don't see why _all_ paths would revert with this.